src/core/tiledscene/qgsesrii3sdataprovider.cpp

```diff
@@ -789,20 +824,27 @@ bool QgsEsriI3SDataProvider::loadFromRestService( const QString &uri, json &laye
 
 bool QgsEsriI3SDataProvider::loadFromSlpk( const QString &uri, json &layerJson, QString &i3sVersion )
 {
-  // Extracted SLPK = SLPK content extracted to a single directory
-
-  // TODO: add true SLPK support (all data packaged in a single ZIP file)
-
-  QUrl rootUrl( uri );
-
-  QString metadataFile = rootUrl.toLocalFile() + "/metadata.json";
-  QFile fMetadata( metadataFile );
-  if ( !fMetadata.open( QIODevice::ReadOnly ) )
+  QByteArray metadataContent;
+  QString metadataFileName = QStringLiteral( "metadata.json" );
+  if ( QFileInfo( uri ).isDir() )  // if a directory, read directly as Extracted SLPK
```

Comment:
in fetchContent(), the detection of SLPK / ESLPK is based on file suffix, here it is based on directory. Maybe at the beginning of loadFromSlpk() we could have a full test:
1. if it is a file with slpk suffix -> SLPK
2. else if it is a directory -> ESLPK
3. else error

and then have just some boolean variable whether we're dealing with eslpk or not...